portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -29,73 +30,87 @@
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  private List<String> messages;
-
-  public void init() {
-      messages = new ArrayList<String>();
-  }
-
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response)
-              throws IOException {
-
-      Query query = new Query("messageEntity//");
-      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-      PreparedQuery results = datastore.prepare(query);
-    
-      for (Entity entity : results.asIterable()) {
-          long id = entity.getKey().getId();
-          String message = (String) entity.getProperty("message");
-          messages.add(message);
-      }
-
-      response.setContentType("application/json;");
-      response.getWriter().println(convertToJsonUsingGson(messages));
-  }
-
-  @Override
-  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-      String clientMessage = request.getParameter("clientMessage");
-      long timeStamp = System.currentTimeMillis();
-
-      Entity messageEntity = new Entity("Messages");
-      messageEntity.setProperty("message", clientMessage);
-      messageEntity.setProperty("timestamp", timeStamp);
-
-      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-      datastore.put(messageEntity);
+    private List<String> messages;
+    private List<Comments> messagesEntityList;
+    private int amtOfComments;
+    private static final String NAME = "name";
+    private static final String EMAIL = "email";
+    private static final String CLIENTMESSAGE = "clientMessage";
+    private static final String TIMESTAMP = "timestamp";
+
+    public void init() {
+        messages = new ArrayList<String>();
+        messagesEntityList = new ArrayList<Comments>();
+        amtOfComments = 0;
+    }
 
-      response.sendRedirect("/index.html");
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response)
+        throws IOException {
+
+        //TO-DO:figure out how to sort queries based on timestamp
+        Query query = new Query("Messages");
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery results = datastore.prepare(query);
+        List<Entity> limitedResults = results.asList(FetchOptions.Builder.withLimit(amtOfComments));
+
+        for (Entity entity : limitedResults) {
+            long id = entity.getKey().getId();
+            String name = (String) entity.getProperty(NAME);
+            String email = (String) entity.getProperty(EMAIL);
+            String message = (String) entity.getProperty(CLIENTMESSAGE);
+            long timeStamp = (long) entity.getProperty(TIMESTAMP);
+
+            Comments commentsObject = new Comments(id, name, email, message, timeStamp);
```

Comment:
Nit: 

1. Class name "Comments" should be singular as it represents a single comment. 
2. you can just name "commentsObject" as "comment". Object suffix doesn't add any value.